service/backend/pchain/types.go

```diff
@@ -42,32 +46,26 @@ func (p *pTx) Unmarshal(bytes []byte) error {
 	if err != nil {
 		return err
 	}
+	if err := tx.Sign(p.Codec, nil); err != nil {
```

Comment:
tx.Sign with nil signer carries out all the initialization a tx needs. I reckon this may not be super apparent from avalanchego codebase inspection. Still we should use it anywhere an initialization is needed